catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/tasks/TaskResource.java

```diff
@@ -0,0 +1,339 @@
+    return Response.created(task.getHref()).entity(task).build();
+  }
+
+  @PATCH
+  @Path("/{id}")
+  @Operation(summary = "Update a Task", tags = "task",
+          description = "Update an existing task using JsonPatch.",
+          externalDocs = @ExternalDocumentation(description = "JsonPatch RFC",
+                  url = "https://tools.ietf.org/html/rfc6902"))
+  @Consumes(MediaType.APPLICATION_JSON_PATCH_JSON)
+  public Task updateDescription(@Context UriInfo uriInfo,
+                                 @Context SecurityContext securityContext,
+                                 @PathParam("id") String id,
+                                 @RequestBody(description = "JsonPatch with array of operations",
+                                         content = @Content(mediaType = MediaType.APPLICATION_JSON_PATCH_JSON,
+                                                 examples = {@ExampleObject("[" +
+                                                         "{op:remove, path:/a}," +
+                                                         "{op:add, path: /b, value: val}" +
+                                                         "]")}))
+                                         JsonPatch patch) throws IOException {
+    Fields fields = new Fields(FIELD_LIST, FIELDS);
+    Task task = dao.get(id, fields);
+    SecurityUtil.checkAdminRoleOrPermissions(authorizer, securityContext,
+            EntityUtil.getEntityReference(task));
+    task = dao.patch(id, patch);
+    return addHref(uriInfo, task);
+  }
+
+  @PUT
+  @Operation(summary = "Create or update task", tags = "tasks",
+          description = "Create a task, it it does not exist or update an existing task.",
+          responses = {
+                  @ApiResponse(responseCode = "200", description = "The updated task ",
+                          content = @Content(mediaType = "application/json",
+                                  schema = @Schema(implementation = Task.class)))
+          })
+  public Response createOrUpdate(@Context UriInfo uriInfo,
+                                 @Context SecurityContext securityContext,
+                                 @Valid CreateTask create) throws IOException {
+
+    Task task =
+            new Task().withId(UUID.randomUUID()).withName(create.getName()).withDisplayName(create.getDisplayName())
+                    .withDescription(create.getDescription())
+                    .withService(create.getService())
+                    .withTaskUrl(create.getTaskUrl())
+                    .withTaskConfig(create.getTaskConfig())
+                    .withTags(create.getTags())
+                    .withOwner(create.getOwner());
+    PutResponse<Task> response = dao.createOrUpdate(task, create.getService(), create.getOwner());
+    task = addHref(uriInfo, response.getEntity());
+    return Response.status(response.getStatus()).entity(task).build();
+  }
+
+  @PUT
+  @Path("/{id}/followers")
```

Comment:
Do we want to allow users to follow tasks or only pipelines?

Reply:
Agree. Doesn't make sense to add follow for tasks